admin/services/class-indexable-post-provider.php

```diff
@@ -0,0 +1,115 @@
+
+		return array(
+			'object_id'                   => (int) $object_id,
+			'object_type'                 => 'post',
+			'object_subtype'              => get_post_type( $object_id ),
+			'permalink'                   => get_permalink( $object_id ),
+			'canonical'                   => $this->get_meta_value( 'canonical', $object_id ),
+			'title'                       => $this->get_meta_value( 'title', $object_id ),
+			'description'                 => $this->get_meta_value( 'metadesc', $object_id ),
+			'breadcrumb_title'            => $this->get_meta_value( 'bctitle', $object_id ),
+			'og_title'                    => $this->get_meta_value( 'opengraph-title', $object_id ),
+			'og_description'              => $this->get_meta_value( 'opengraph-description', $object_id ),
+			'og_image'                    => $this->get_meta_value( 'opengraph-image', $object_id ),
+			'twitter_title'               => $this->get_meta_value( 'twitter-title', $object_id ),
+			'twitter_description'         => $this->get_meta_value( 'twitter-description', $object_id ),
+			'twitter_image'               => $this->get_meta_value( 'twitter-image', $object_id ),
+			'is_robots_noindex'           => $this->translate_robots_noindex( $this->get_meta_value( 'meta-robots-noindex', $object_id ) ),
+			'is_robots_nofollow'          => $this->get_meta_value( 'meta-robots-nofollow', $object_id ) === '1',
+			'is_robots_noarchive'         => in_array( 'noarchive', $meta_robots_adv, true ),
+			'is_robots_noimageindex'      => in_array( 'noimageindex', $meta_robots_adv, true ),
+			'is_robots_nosnippet'         => in_array( 'nosnippet', $meta_robots_adv, true ),
+			'primary_focus_keyword'       => $this->get_meta_value( 'focuskw', $object_id ),
+			'primary_focus_keyword_score' => (int) $this->get_meta_value( 'linkdex', $object_id ),
+			'readability_score'           => (int) $this->get_meta_value( 'content_score', $object_id ),
+			'is_cornerstone'              => $this->get_meta_value( 'is_cornerstone', $object_id ) === '1',
+			'link_count'                  => (int) $link_count->get( $object_id ),
+			'incoming_link_count'         => (int) $link_count->get( $object_id, 'incoming_link_count' ),
+			'created_at'                  => null,
+			'updated_at'                  => null,
+		);
+	}
+
+	/**
+	 * Checks if the given object id belongs to an indexable.
+	 *
+	 * @param int $object_id The object id.
+	 *
+	 * @return bool Whether the object id is indexable.
+	 */
+	public function is_indexable( $object_id ) {
+		if ( get_post( $object_id ) === null ) {
+		 	return false;
+		}
+
+		if ( wp_is_post_autosave( $object_id ) ) {
+			return false;
+		}
+
+		if ( wp_is_post_revision( $object_id ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Translates the meta value to a boolean value.
+	 *
+	 * @param string $value The value to translate.
+	 *
+	 * @return bool|null The translated value.
+	 */
+	protected function translate_robots_noindex( $value ) {
```

Comment:
I don't think `translate` is the best term for this.
Would think the naive `get` would be better, or `transform` but that would need a from-to structure.

Suggested: `get_robots_noindex_value`